apps/prairielearn/src/pages/elementFiles/elementFiles.js

```diff
@@ -23,64 +22,76 @@ const router = Router({ mergeParams: true });
 const EXTENSION_WHITELIST = ['.js', '.css'];
 const CLIENT_FOLDER = 'clientFilesElement';
 
-router.get(
-  '/*',
-  asyncHandler(async (req, res) => {
-    const filename = req.params[0];
-    const pathSpl = path.normalize(filename).split('/');
-    const valid =
-      pathSpl[1] === CLIENT_FOLDER ||
-      EXTENSION_WHITELIST.some((extension) => filename.endsWith(extension));
-    if (!valid) {
-      throw new HttpStatusError(404, 'Unable to serve that file');
-    }
+export default function (options = { publicEndpoint: false, static: false }) {
```

Comment:
I'm not a fan of this structure, it deviates from other pages. I'd suggest have this be a named (non-default) function, and have the default be a const corresponding to the result of calling this function with no arguments.

Reply:
This is the same structure that I used for the clientFilesCourse endpoint at the request of @nwalters512 : 

https://github.com/PrairieLearn/PrairieLearn/blob/0f34dd9ef80a69f05c2d115855aab30a40f05abb/apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js#L10-L15

Reply:
I do like Jonatan's suggestion. I'd be happy to move in that direction, but I'd rather either a) land that change first or b) land this PR and then change everything at once.

Reply:
I'd like to get this PR in because we've had complaints already about the feature not working, so I'm not going to worry about this for now.